toml/decoder.py

```diff
@@ -75,8 +75,12 @@ def _strictly_valid_num(n):
         return False
     if len(n) == 1:
         return True
-    if n[0] == '0' and n[1] != '.':
-        return False
+    if n[0] == '0':
+        if n[1] in ['o','b','x']:
+            n = int(n, 0)
+            return True
+        if n[1] != '.':
+            return False
```

Comment:
I think this should just be `if n[0] == '0' and n[1] not in ['.','o','b','x']` then `return False`
I am concerned that `int(n, 0)` might cause a ValueError to bubble up where it shouldn't.

Reply:
I also agree that my approach looks a bit ugly but I wanted to avoid passing floats to int(n, 0).

Reply:
Forget about this ^ I just realised how stupid my comment is